### PR TITLE
Support initial balance allocations

### LINKED ISSUE #77: Support initial stx balance
Currently, the VM is started with 0 balance for all accounts. (I think)

For tests, it would be beneficial if the initial balance could be set.

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds a new parameter `initialAllocations` to the `NativeClarityBinProvider`
* adds a test for initial stx balance allocation
* fixes #77

### CODE REVIEW

File: packages/clarity/test/nativeClarityBinProvider.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { assert } from "chai";
+import fs from "fs-extra";
+import { NativeClarityBinProvider, Receipt } from "../src";
+import { getTempFilePath } from "../src/utils/fsUtil";
+
+describe("NativeClarityBinProvider", () => {
+  it("create ephemeral", async () => {
+    const provider = await NativeClarityBinProvider.createEphemeral(
+      "node_modules/@blockstack/clarity-native-bin/.native-bin/clarity-sdk-v0.1.0/clarity-cli"
```
Comment (friedger) [inline packages/clarity/test/nativeClarityBinProvider.ts]: What should be the correct path for the latest clarity-cli ?

File: packages/clarity/src/providers/clarityBin/index.ts [hunk 2]
```diff
@@ -69,7 +72,7 @@ export class NativeClarityBinProvider implements Provider {
    */
   async runCommand(args: string[], opts?: { stdin: string }) {
     const result = await executeCommand(this.clarityBinPath, [...args], {
-      stdin: opts && opts.stdin
+      stdin: opts && opts.stdin,
```
Comment (friedger) [inline packages/clarity/src/providers/clarityBin/index.ts:75]: these style changes should be fixes by #102

File: packages/clarity-native-bin/src/index.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { ConsoleLogger, ILogger } from "./logger";
  * Should correspond to both a git tag on the blockstack-core repo and a
  * set of clarity-binary distributables uploaded to the cloud storage endpoint.
  */
-export const CORE_SDK_TAG = "clarity-sdk-v0.1.0";
+export const CORE_SDK_TAG = "clarity-sdk-v0.1.1";
```
Comment (friedger) [inline packages/clarity-native-bin/src/index.ts:12]: This needs to be higher than 0.1.1!

File: packages/clarity/src/providers/clarityBin/index.ts [hunk 4]
```diff
@@ -91,7 +97,7 @@ export class NativeClarityBinProvider implements Provider {
         result.stderr
       );
     }
-    if (result.stdout !== "Database created.") {
+    if (!result.stdout.endsWith("Database created.")) {
```
Comment (friedger) [inline packages/clarity/src/providers/clarityBin/index.ts:100]: now we have more output from the initial allocations

File: packages/clarity/test/nativeClarityBinProvider.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      tempDbPath,
+      "node_modules/@blockstack/clarity-native-bin/.native-bin/clarity-sdk-v0.1.0/clarity-cli"
+    );
+
+    const receipt = await getStxBalance(principal, provider);
+    assert.equal(receipt.result, `u${amount.toString()}`);
+  });
+});
+
+async function getStxBalance(
```
Comment (friedger) [inline packages/clarity/test/nativeClarityBinProvider.ts:29]: Is there a better way to check the balance than creating an empty contract?

**Comment by zone117x** [comment]
This is great @friedger!
Can you rebase onto master? Binaries are fixed and tests should now pass

File: packages/clarity/test/nativeClarityBinProvider.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { assert } from "chai";
+import fs from "fs-extra";
+import { NativeClarityBinProvider, Receipt } from "../src";
+import { getTempFilePath } from "../src/utils/fsUtil";
+
+describe("NativeClarityBinProvider", () => {
+  it("create ephemeral", async () => {
+    const provider = await NativeClarityBinProvider.createEphemeral(
```
Comment (zone117x) [inline packages/clarity/test/nativeClarityBinProvider.ts:8]: `NativeClarityBinProvider.createEphemeral(getDefaultBinaryFilePath())` should work once this is rebased off latest master

File: packages/clarity/test/nativeClarityBinProvider.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  it("set initial allocations", async () => {
+    const principal = "ST12EY99GS4YKP0CP2CFW6SEPWQ2CGVRWK5GHKDRV";
+    const amount = 1000;
+    const tempDbPath = getTempFilePath("blockstack-local-{uniqueID}.db");
+    const provider = await NativeClarityBinProvider.create(
+      [{ principal, amount }],
+      tempDbPath,
+      "node_modules/@blockstack/clarity-native-bin/.native-bin/clarity-sdk-v0.1.0/clarity-cli"
```
Comment (zone117x) [inline packages/clarity/test/nativeClarityBinProvider.ts:21]: Same here `getDefaultBinaryFilePath()`

File: packages/clarity/test/nativeClarityBinProvider.ts (hunk 5 above)
Comment (zone117x) [reply packages/clarity/test/nativeClarityBinProvider.ts:29]: I think the `eval_raw` command would need modified to support passing in a db file

**Comment by friedger** [comment]
@zone117x The test still fails because clarity 0.1.1 is not up to date enough, see https://github.com/blockstack/stacks-blockchain/issues/1833

File: packages/clarity/test/nativeClarityBinProvider.ts (hunk 5 above)
Comment (friedger) [reply packages/clarity/test/nativeClarityBinProvider.ts:30]: the documentation says that no db context is used for `eval_raw`. Is the empty contract a good solution?

**Comment by zone117x** [comment]
Try with latest master now https://github.com/blockstack/clarity-js-sdk/pull/103

**Comment by friedger** [comment]
@psq Has been removed in 2e438ca0304222ae20826c068b4f7a47532469c9

**Comment by psq** [comment]
thank you for doing this, @friedger.  A lot of people will thank you!

LGTM

File: packages/clarity/test/nativeClarityBinProvider.ts (hunk 5 above)
Comment (zone117x) [reply packages/clarity/test/nativeClarityBinProvider.ts:30]: The rust code for `eval_raw` would ideally be updated to support a db context. This works for now though

**Pull request merged**